core/domain/question_domain.py

```diff
@@ -1503,7 +1503,9 @@ def partial_validate(self) -> None:
             if answer_group.outcome.dest_if_really_stuck is not None:
                 dest_if_stuck_is_specified = True
 
-        # Ruling out the possibility of None for mypy type checking.
+        # Ruling out the possibility of None for mypy type checking, because
```

Comment:
If this is the case then why isn't default_outcome _always_ not None? Based on this argument we should have non-None as part of the type.

Reply:
because in exploration `default_outcome` can be None
https://github.com/oppia/oppia/blob/4ba04d87b7e8aea3953287ae183c8e7d1c0031b6/core/domain/exp_domain.py#L1043
this default_outcome passed to interaction class
https://github.com/oppia/oppia/blob/4ba04d87b7e8aea3953287ae183c8e7d1c0031b6/core/domain/exp_domain.py#L1060


> Based on this argument we should have non-None as part of the type.

Not possible because both question and exploration shares the same state_domain's interaction class. So, to fulfill the cases of exploration we have to mark `default_outcome` as None.

Reply:
OK, please explain that in your comment.

Reply:
I meant in the code -- "Note that interaction.default_outcome can be None in the case of explorations but not for questions, which is why we need the assertion here." or something like that.

Reply:
comment changed